### PR TITLE
[refactor] Some automatic refactoring using IntelliJ

### PR DESCRIPTION
**Pull request by dizzzz**
Some automatic refactoring using IntelliJ: use java 7/8/14/16 language features.

Grouped per commit.

### CODE REVIEW

**Comment by reinhapa** [comment]
> @reinhapa yeah I spotted them, I will check if they were not already there.

Seems only one class being the issue, that you may can change with low effort...

**Comment by dizzzz** [comment]
there are 3 classes, all previously risking the NPE :-(

the largest I could solve by simply not returning the null value, but the already defined QName.EMPTY_QNAME

```
@Override
    public final QName getQName() {
        return switch (getNodeType()) {
            case Node.ATTRIBUTE_NODE -> document.attrName[nodeNumber];
            case Node.ELEMENT_NODE, Node.PROCESSING_INSTRUCTION_NODE -> document.nodeName[nodeNumber];
            case NodeImpl.NAMESPACE_NODE -> document.namespaceCode[nodeNumber];
            case Node.DOCUMENT_NODE -> QName.EMPTY_QNAME;
            case Node.COMMENT_NODE -> QName.EMPTY_QNAME;
            case Node.TEXT_NODE -> QName.EMPTY_QNAME;
            case Node.CDATA_SECTION_NODE -> QName.EMPTY_QNAME;
            default -> null;
        };
    }
```

looks not to be of a problem right here, but it might be downstream.

for the other two... few... @adamretter @wolfgangmm  ?

**Comment by adamretter** [comment]
> for the other two... pfew... @adamretter 

@dizzzz I see 5 listed. Which of those are the one or two you would like me to take a look at?

**Comment by dizzzz** [comment]
so there are 3 identical situations in NodeImpl.java, where we could just return the 'null' Qname
for the other 2.... no idea.

File: exist-core/src/main/java/org/exist/dom/memtree/NodeImpl.java [hunk 1]
```diff
@@ -151,21 +139,21 @@ public String getLocalName() {
...
                     return nsUri;
                 }
-
-            case NodeImpl.NAMESPACE_NODE:
+            }
+            case NodeImpl.NAMESPACE_NODE -> {
                 return XMLConstants.XMLNS_ATTRIBUTE_NS_URI;
-
-            default:
+            }
```
Comment (reinhapa) [inline exist-core/src/main/java/org/exist/dom/memtree/NodeImpl.java:153]: `case NodeImpl.NAMESPACE_NODE -> XMLConstants.XMLNS_ATTRIBUTE_NS_URI;`

File: exist-core/src/main/java/org/exist/dom/memtree/NodeImpl.java [hunk 2]
```diff
@@ -151,21 +139,21 @@ public String getLocalName() {
...
+            default -> {
                 return null;
+            }
```
Comment (reinhapa) [inline exist-core/src/main/java/org/exist/dom/memtree/NodeImpl.java:156]: `default -> null;`

**Review by reinhapa — CHANGES_REQUESTED** [review]
Some small parts that can be simplified even more...

**Review by reinhapa — APPROVED** [review]

**Review by reinhapa — APPROVED** [review]

**Review by line-o — APPROVED** [review]
I like the changes :tada:

**Review by reinhapa — COMMENTED** [review]
@dizzzz getting better every day 💪

**Pull request merged**